### DISCUSSION TITLE
How to highlight the substructure  in the image using Fragment module

### DISCUSSION THREAD

**Discussion by gundalav** [rdkit/rdkit#4648] [General] [answered]
I have the following code that highlight the carbonyl group on my molecule:

```
from rdkit import Chem
from rdkit.Chem.Draw import IPythonConsole
from rdkit.Chem import rdDepictor
from rdkit.Chem.Draw import rdMolDraw2D
from IPython.display import SVG


def increase_resolution(mol, substructure, size=(400, 200)):
    rdDepictor.Compute2DCoords(mol)
    Chem.Kekulize(mol) # Localize the benzene ring bonds
    
    drawer = rdMolDraw2D.MolDraw2DSVG(size[0], size[1])
    
    # highlightAtoms expects only one tuple, not tuple of tuples. So it needs to be merged into a single tuple
    matches = sum(mol.GetSubstructMatches(substructure), ())
    drawer.DrawMolecule(mol, highlightAtoms=matches)
    
    drawer.FinishDrawing()
    svg = drawer.GetDrawingText()
    
    return SVG(svg.replace('svg:',''))


mol = Chem.MolFromSmiles('c1cc(C(=O)O)c(OC(=O)C)cc1')
substructure = Chem.MolFromSmarts('C=O')
increase_resolution(mol, substructure)
```

Which produce this image:
<img width="249" alt="Screen Shot 2021-10-27 at 21 12 25" src="https://user-images.githubusercontent.com/8143832/139063142-a8a81c52-2d75-4138-b120-cafe86bac473.png">

Note that I manually hard code the carbonyl group.
`Chem.MolFromSmarts('C=O')`

RDKit provides the [**Fragment module**](http://rdkit.org/docs/source/rdkit.Chem.Fragments.html) to identify substructure in the molecule. With function that looks like:

```
rdkit.Chem.Fragments.fr_C_O()
rdkit.Chem.Fragments.fr_Al_COO()
..etc..
```

Is there a way to highlight the image using  [**Fragment module**](http://rdkit.org/docs/source/rdkit.Chem.Fragments.html) ?

**Comment by dehaenw** [comment, marked as answer]
If I'm reading the documentation right, I think the Fragment module only returns the number of matches, and no details about where those matches exactly are in the molecule. In case you need a quick reference that has all the SMARTS used in the fragment module uses you can check here https://github.com/rdkit/rdkit/blob/master/Data/FragmentDescriptors.csv

**Reply by gundalav** [reply]
Thank you. That'll do.
